From 193543f976e6a5857175eecfe192b68c305bf752 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd.simons@collabora.co.uk>
Date: Fri, 23 Sep 2016 23:12:56 +0200
Subject: [PATCH] wayland/xdg-shell: update popup window monitor early

As meta_window_place_with_placement_rule will trigger a configure event
being sent ensure that the popup is placed on the correct monitor first
to ensure the right scale factor is applied.

Signed-off-by: Sjoerd Simons <sjoerd.simons@collabora.co.uk>

https://bugzilla.gnome.org/show_bug.cgi?id=771841
---
 src/wayland/meta-wayland-xdg-shell.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/wayland/meta-wayland-xdg-shell.c b/src/wayland/meta-wayland-xdg-shell.c
index b1b3771..4e572dc 100644
--- a/src/wayland/meta-wayland-xdg-shell.c
+++ b/src/wayland/meta-wayland-xdg-shell.c
@@ -793,8 +793,9 @@ finish_popup_setup (MetaWaylandXdgPopup *xdg_popup)
                                     &xdg_popup->parent_destroy_listener);
 
   window = meta_window_wayland_new (display, surface);
-  meta_window_place_with_placement_rule (window, &placement_rule);
   meta_wayland_surface_set_window (surface, window);
+  meta_window_update_monitor (window, FALSE);
+  meta_window_place_with_placement_rule (window, &placement_rule);
 
   if (seat)
     {
-- 
2.9.3

